ElenaCzubiak-elena-cz/HW_6/BookManager2/app/controllers/books_controller.rb

```diff
@@ -0,0 +1,51 @@
+class BooksController < ApplicationController
+    
+    before_action :authenticate_user!
+    
```

Comment:
Nifty tip about the `ApplicationsController`: since all your other controllers are descendants, you can DRY out your code by including things you'd want in all your controllers like `before_action :authenticate_user!`

Try adding `before_action :authenticate_user!` to your `ApplicationsController` and remove it from your other controllers.